src/Plugin/rest/resource/AlertsRestResource.php

```diff
@@ -195,11 +208,33 @@ public function get() {
     }
 
     $uri = $this->request->query->get('uri');
+
+    // Extract the language code from the URI.
+    $langcode = substr($uri, 1, 2);
+    $language = $this->languageManager->getLanguage($langcode);
+    $alert_language = NULL;
+    if ($language) {
+      $alert_language = $language->getId();
+    }
+
+    // Check if the first segment is a valid language code.
+    $request_langcode = substr($this->request->getPathInfo(), 1, 2);
+    $request_language = $this->languageManager->getLanguage($request_langcode);
+
+    if ($language && !$request_language || $request_language && $request_language->getId() != $alert_language) {
+      // Redirect to the same request URL with the language code added after the basePath.
+      $redirectUrl = $this->request->getSchemeAndHttpHost() . '/' . $langcode . $this->request->getRequestUri();
+      $response = new ModifiedResourceResponse(NULL, 303); // 303 See Other
```

Comment:
Let's stick to Drupal code style, please

Reply:
This is resolved.